File: app/src/main/java/com/example/notesync/AmigosAdapter.java
package com.example.notesync;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class AmigosAdapter extends RecyclerView.Adapter<AmigosAdapter.AmigoViewHolder> {
    private List<Usuario> amigos;

    public AmigosAdapter(List<Usuario> amigos) {
        this.amigos = amigos;
    }

    @NonNull
    @Override
    public AmigoViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(android.R.layout.simple_list_item_1, parent, false);
        return new AmigoViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull AmigoViewHolder holder, int position) {
        holder.nome.setText(amigos.get(position).getNome());
    }

    @Override
    public int getItemCount() {
        return amigos.size();
    }

    static class AmigoViewHolder extends RecyclerView.ViewHolder {
        TextView nome;
        AmigoViewHolder(View itemView) {
            super(itemView);
            nome = itemView.findViewById(android.R.id.text1);
        }
    }
}

File: app/src/main/java/com/example/treino1/AmigosAdapter.kt
package com.example.treino1

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class AmigosAdapter(
    private val amigos: List<UsuarioCompartilhado>,
    private val onRemoverClick: (UsuarioCompartilhado) -> Unit
) : RecyclerView.Adapter<AmigosAdapter.AmigoViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AmigoViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_amigo_compartilhado, parent, false)
        return AmigoViewHolder(view)
    }

    override fun onBindViewHolder(holder: AmigoViewHolder, position: Int) {
        val amigo = amigos[position]
        holder.nome.text = amigo.nome
        holder.id.text = "ID: ${amigo.idUsuario}"
        holder.btnRemover.setOnClickListener {
            onRemoverClick(amigo)
        }
    }

    override fun getItemCount() = amigos.size

    class AmigoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nome: TextView = itemView.findViewById(R.id.textNomeAmigo)
        val id: TextView = itemView.findViewById(R.id.textIdAmigo)
        val btnRemover: ImageView = itemView.findViewById(R.id.btnRemoverAmigo)
    }
}
